docs/developer-guide/api-overview.md

```diff
@@ -20,153 +20,155 @@ The relevant modules are available in Maven Central:
+that provide the HTTP headers and the full body in one immutable unit.
+For this purpose Styx provides `HttpRequest` and `HttpResponse` classes. They are immutable, 
+fully re-usable HTTP message classes that are more convenient for unit testing, admin interfaces,
+and for other applications built on Styx libraries.
 
-Few words about message interoperability. The streaming and full variants are not 
+A few words about message interoperability: the live and immutable variants are not 
 interface compatible as per
 [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) 
-and therefore they form two separate class hierarchies. The nature of the content,
-streaming vs full, is reflected in the type signatures of the content accessors:
+and therefore they form two separate class hierarchies. This is evident in the
+HTTP content accessors:
 
-* Streaming HTTP messages: 
+* `LiveHttpRequest` (and `LiveHttpResponse`): 
 
 ```java
-    public class HttpRequest { 
+    public class LiveHttpRequest { 
         ...
         
-        public StyxObservable<ByteBuf> body() { .. }
+        public ByteStream body() { .. }
     }
 ```
 
-* Full HTTP messages:
+* `HttpRequest` (and `HttpResponse`)
    
 ```java
-    public class FullHttpRequest { 
+    public class HttpRequest { 
         ...       
         public byte[] body() { .. }
         
         public String bodyAs(Charset charset) { .. }        
     }
 ```
 
-### Conversion Between HTTP Message Types
- 
-It is easy to convert between the streaming and full HTTP messages.  
+### Conversion Between HTTP Message Types 
 
-Streaming `HttpRequest` has a `toFullRequest` method (`toFullResponse` for `HttpResponse`).
-It aggregates a HTTP message body stream into one continuous byte array and creates a 
-corresponding `FullHttpRequest` (`FullHttpResponse`). Its type signature is:
+Call the `aggregate` method to convert a `LiveHttpRequest` (`LiveHttpResponse`) 
+to an immutable `HttpRequest` (`HttpResponse`) object: 
 
 ```java
-   public StyxObservable<FullHttpRequest> toFullRequest(int maxContentBytes);
+   public Eventual<HttpRequest> aggregate(int maxContentBytes);
 ```
 
-Note that the HTTP content streams represents live network traffic streaming
-through Styx core. The content stream is not memoized nor stored. 
-When the content is gone, it is gone for good. 
-Therefore a `HttpRequest` (`HttpResponse`) can be aggregated to a full 
-message strictly once only.
+This aggregates a HTTP message body stream and creates a corresponding immutable 
+HTTP object. Note that `aggregate` consumes the underlying live HTTP message `ByteStream`.
 
 The type signature illustrates other facts about the content stream:
 
-* Return type of `StyxObservable` shows that HTTP content stream is 
-  aggregated asynchronously. Obviously because the rest of the content 
-  stream are yet to be received from the network.
+* Return type of `Eventual` shows that `aggregate` is asynchronous. 
```

Comment:
This line is redundant. We should delete this line ("Return type of ....") and keep the one below ("We can see that....")

Reply:
👍